huaweicloud/services/rds/resource_huaweicloud_rds_instance.go

```diff
@@ -872,13 +898,48 @@ func resourceRdsInstanceDelete(ctx context.Context, d *schema.ResourceData, meta
 	id := d.Id()
 	log.Printf("[DEBUG] Deleting Instance %s", id)
 	if v, ok := d.GetOk("charging_mode"); ok && v.(string) == "prePaid" {
-		if err := common.UnsubscribePrePaidResource(d, config, []string{id}); err != nil {
+		resourceIds := []string{id}
+		// the image of SQL server is come from cloud market, when creating an SQL server instance resource, two order
+		// will be created, one is instance order, the other is market image order, so it is needed to unsubscribe the
+		// two order when unsubscribe the instance
+		if strings.ToLower(d.Get("db.0.type").(string)) == "sqlserver" {
+			resourceIds = append(resourceIds, fmt.Sprintf("%s%s", id, ".marketimage"))
+		}
+		retryFunc := func() (interface{}, bool, error) {
+			err = common.UnsubscribePrePaidResource(d, config, resourceIds)
+			log.Printf("UnsubscribePrePaidResource err: %s", err)
+			retry, err := handleDeletionError(err)
+			return nil, retry, err
+		}
+		_, err = common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+			Ctx:          ctx,
+			RetryFunc:    retryFunc,
+			WaitFunc:     rdsInstanceStateRefreshFunc(client, id),
+			WaitTarget:   []string{"ACTIVE"},
```

Comment:
```suggestion
			WaitTarget:   []string{"DELETED"},
```

Reply:
when deleting instance, if the instance status is not active, an error will be occured, so it is needed to wait the statue of the instance to become active, this retry func is used to wait for the status of the instance to become active